blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/credential/CredentialApi.java

```diff
@@ -88,6 +94,32 @@ public CreateResponse create(@JsonBody JSONObject body, StaplerRequest request)
         throw new ServiceException.UnexpectedErrorException("Unexpected error, failed to create credential");
     }
 
+    @GET
+    @WebMethod(name="publickey")
+    public void getPersonalKey(StaplerResponse rsp) throws IOException {
```

Comment:
@vivek This is a different API, it automatically generates a special private key for the user that only Jenkins has access to.